### ISSUE TITLE
Main Pebble Quail - Calling GovernorBravoDelegate::queue() function on existing proposal across blocks will result in invalid transaction records queued in Timelock contract

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2024-01-olympus-on-chain-governance-judging#44]
Main Pebble Quail

medium

# Calling GovernorBravoDelegate::queue() function on existing proposal across blocks will result in invalid transaction records queued in Timelock contract

## Summary
queue() is an external function, which can be called for a pending proposal. An attacked can successfully insert invalid transaction records into the transaction queue of the time locker contract.

This breaks the data synergy between the GovernorBravoDelegate() and timelock contract.

## Vulnerability Detail
The queue() function checks if the proposal id exists  and did not succeed yet.
Every time, the queue function is called, the **eta** is updated as below. Also, note how **proposal.eta** is also updated every time the queue is called.

```solidity
        Proposal storage proposal = proposals[proposalId];
        uint256 eta = block.timestamp + timelock.delay();
        ...supressed code
        proposal.eta = eta;
```
So, lets say the first time the queue is called, the eta was
    eta = 1705999212 + 86400 * 2(2 days)
    eta = 1706172012

After few blocks, when the eta will be computed, it will be different.
   eta = 1706172012 + 3600
         = 1706175612

Now, the issue is that the **eta is part of the hash generation data for each transaction**. As such, for the same proposal, different transaction records are queued into the timelock contract.

But for processing, again the hash is generated using the **latest eta** and hence many transactions inserted into the timelock contract will remain in **queued state and will remain delink from their proposals** in GovernorBravoDelegate contract.

## Impact
generation of invalid transactions in timelock contract.

## Code Snippet
queue() function which calls _queueOrRevertInternal() function for each target in the proposal.
Note, how the eta is passed to the _queueOrRevertInternal() function.
https://github.com/sherlock-audit/2024-01-olympus-on-chain-governance/blob/main/bophades/src/external/governance/GovernorBravoDelegate.sol#L211-L236

Note, how **eta is part of the hash generated** for the target and hence will result in a **totally different hash**, every time
the queue call is made, the requirement is that calls are spread across blocks to ensure new hashs.

https://github.com/sherlock-audit/2024-01-olympus-on-chain-governance/blob/main/bophades/src/external/governance/GovernorBravoDelegate.sol#L238-L249

## Tool used

Manual Review

## Recommendation
if proposal.eta is not 0, then assume that the proposal was already queued and revert the queue function.

**Comment by 0xLienid** [comment]
this is not true. `state` already handles marking proposals as queued after the first call
